File: dbflow/src/main/java/com/raizlabs/android/dbflow/structure/database/DatabaseStatement.java
package com.raizlabs.android.dbflow.structure.database;

import android.database.sqlite.SQLiteStatement;
import android.support.annotation.Nullable;

/**
 * Description: Abstracts out a {@link SQLiteStatement}.
 */
public interface DatabaseStatement {

    long executeUpdateDelete();

    void execute();

    void close();

    long simpleQueryForLong();

    String simpleQueryForString();

    long executeInsert();

    void bindString(int index, String s);

    void bindStringOrNull(int index, @Nullable String s);

    void bindNull(int index);

    void bindLong(int index, long aLong);

    void bindNumber(int index, @Nullable Number number);

    void bindNumberOrNull(int index, @Nullable Number number);

    void bindDouble(int index, double aDouble);

    void bindDoubleOrNull(int index, @Nullable Double aDouble);

    void bindBlob(int index, byte[] bytes);

    void bindBlobOrNull(int index, @Nullable byte[] bytes);

}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/database/DatabaseStatement.kt
package com.raizlabs.android.dbflow.database

import android.database.sqlite.SQLiteStatement

/**
 * Description: Abstracts out a [SQLiteStatement].
 */
interface DatabaseStatement {

    fun executeUpdateDelete(): Long

    fun execute()

    fun close()

    fun simpleQueryForLong(): Long

    fun simpleQueryForString(): String?

    fun executeInsert(): Long

    fun bindString(index: Int, s: String)

    fun bindStringOrNull(index: Int, s: String?)

    fun bindNull(index: Int)

    fun bindLong(index: Int, aLong: Long)

    fun bindNumber(index: Int, number: Number?)

    fun bindNumberOrNull(index: Int, number: Number?)

    fun bindDouble(index: Int, aDouble: Double)

    fun bindDoubleOrNull(index: Int, aDouble: Double?)

    fun bindFloatOrNull(index: Int, aFloat: Float?)

    fun bindBlob(index: Int, bytes: ByteArray)

    fun bindBlobOrNull(index: Int, bytes: ByteArray?)

}
